File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    vowels = ['A', 'E', 'I', 'O', 'U']
    # 각 자릿수별로 가능한 조합의 수. 누적합
    place_value = [781, 156, 31, 6, 1]
    answer = 0

    for i in range(len(word)):
        # 각 문자의 위치와 그 문자가 있는 자릿수별 가중치를 곱하고
        # 마지막으로 현재 자릿수까지 포함시키기 위해 +1
        answer += vowels.index(word[i]) * place_value[i] + 1

    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
  const vowels = ["A", "E", "I", "O", "U"];
  const MAX_LENGTH = 5;
  const placeValues = getPlaceValues(MAX_LENGTH, vowels.length);
  let answer = 0;

  for (let i = 0; i < word.length; i++) {
    answer += vowels.indexOf(word[i]) * placeValues[i] + 1;
  }

  return answer;
}

function getPlaceValues(length, caseCount) {
  const placeValues = [1];
  for (let i = 1; i < length; i++) {
    const combCounts = placeValues[i - 1] + Math.pow(caseCount, i);
    placeValues.push(combCounts);
  }

  return placeValues.reverse();
}
